feat(protocol-designer, step-generation, components, shared-data): add flex and move labware

b-cooper opened the pull request:
# Overview

Introduce a new file creation wizard within protocol designer that sequentially walks users through
initializing a protocol for either the OT2 or the Opentrons Flex. Plug these new values into the
rest of the application's state, and follow the path all the way to the analysis/protocol engine via
updates to step generation that allow it to handle moving labware and speccing a robot type inside of a JSON
protocol file.

 Use modern module visualization components for rendering deck map items now that the
deck definition is variable within protocol designer. Render timeline module state visually on the
module svg with a new generic module name label. Remove step-generation error handling for deck
collisions that don't apply to the Flex. Present the Magnetic Block on the File Info Modules
section.

# Test Plan

- create a Flex protocol in PD
- create an OT2 protocol in PD

# Risk assessment
medium, this has a large footprint

b-cooper committed:
move labware form built

b-cooper committed:
labware location dropdown wiredup

b-cooper committed:
wire up to command creator

b-cooper committed:
helpers for deck state

b-cooper committed:
add dynamic labware location to deck setup component

b-cooper committed:
add move labware support to json translator and step generation

b-cooper committed:
get next robot state and warning for move labware with location object

b-cooper committed:
dont use connect anymore

b-cooper committed:
move labware hardcode new deck def for now

b-cooper committed:
robot type reducer scaffold

b-cooper committed:
robot type in global state

b-cooper committed:
pipette select

b-cooper committed:
scaffold create file wizard

b-cooper committed:
creat file wizard

b-cooper committed:
create file wizard mvp with steps

b-cooper committed:
split out choose robot type

b-cooper committed:
make local wizard header and pull in react i18next

b-cooper committed:
react i18next and formatting

b-cooper committed:
save game

b-cooper committed:
pipette select GENERIC

b-cooper committed:
need to fix up handle submit

b-cooper committed:
handle create file button

b-cooper committed:
fix up a bit of alignment and design note

b-cooper committed:
module setup for flex

b-cooper committed:
emily notes on module select

b-cooper committed:
module select form fix

b-cooper committed:
set slots for modules from form

jerader committed:
remove unnecessary error creators for Flex

jerader committed:
filter out 96-channel for now

b-cooper committed:
real module viz

b-cooper committed:
fix description and org mapping

b-cooper committed:
correct tip vol mapping

jerader committed:
fix metadataTile and display magnetic block properly

b-cooper committed:
split out tip and pipette type select

jerader committed:
hide move location step creation option for ot-2

jerader committed:
Revert "hide move location step creation option for ot-2"

This reverts commit df595a8c78022cad7bb3f2f9d3108552a1996c47.

jerader commented on api/src/opentrons/protocol_runner/json_translator.py:
```diff
@@ -136,6 +136,7 @@ def translate_commands(
         """Takes json protocol v6 and translates commands->protocol engine commands."""
         commands_list: List[pe_commands.CommandCreate] = []
         for command in protocol.commands:
+            print(f'\n\n\n\n\n\n{command}\n\n\n\n')
```
what's this? @b-cooper

b-cooper committed:
position nested labware directly from module generics

jerader committed:
fix lint and some frontend unit tests

jerader committed:
remove empty space at the end of the file name

jerader committed:
remove print from json_translator and delete duplicate tiprack flex defs

b-cooper committed:
Merge branch 'pd_gripper-support' of github.com:Opentrons/opentrons into pd_gripper-support

b-cooper committed:
thermocycler inner props cleanup

b-cooper committed:
block based select for pipette and tip

b-cooper committed:
matrix multiplication for module offsets

b-cooper committed:
grab robot type on load file

jerader committed:
fix lint and update snaps

jerader committed:
fix check errors, delete cypress tests

b-cooper committed:
fix ot2 tc rendering

b-cooper committed:
handle enter to proceed, shift enter for new line in text area

b-cooper committed:
Merge branch 'pd_gripper-support' of github.com:Opentrons/opentrons into pd_gripper-support

b-cooper committed:
format

b-cooper committed:
revert unnecessary changes

b-cooper committed:
remove unused new file modal

b-cooper commented on shared-data/js/types.ts:
```diff
@@ -324,11 +324,7 @@ export interface ModuleDefinition {
   twoDimensionalRendering: INode
 }
 
-export type AffineTransformMatrix = [
```
The matrix is actually 4x4

b-cooper commented on step-generation/src/commandCreators/atomic/replaceTip.ts:
```diff
@@ -103,26 +106,28 @@ export const replaceTip: CommandCreator<ReplaceTipArgs> = (
-  ) {
-    return { errors: [errorCreators.heaterShakerEastWestWithLatchOpen()] }
-  }
+  if (!isFlexPipette) {
```
This logic is fine, but hopefully the errors will be more of a "OT2 only check", it would therefore be a little more future proof here to say `if (OT2)` compared to `if (!Flex)`

jerader replied:
yeah, that makes sense. I was having a hard time finding how to get the robot type and pipettes was already there so figured it was fine to filter based on pipettes. But happy to change it to checking robot type in a follow up pr

b-cooper committed:
feat(protocol-designer, step-generation, components, shared-data): add flex and move labware

Introduce a new file creation wizard within protocol designer that sequentially walks users through
initializing a protocol for either the OT2 or the Opentrons Flex. Plug these new values into the
rest of the applications state, and follow the path all the way to the analysis/protocol engine via
updates to step generation to handle moving labware and speccing a robot type inside of a JSON
protocol file. Use modern module visualization components for rendering deck map items now that the
deck definition is variable within protocol designer. Render timeline module state visually on the
module svg with a new generic module name label. Remove step-generation error handling for deck
collisions that don't apply to the Flex. Present the Magnetic Block on the File Info Modules
section.

jerader reviewed (approved):
🤩 woo!
